File: 프로그래머스/2/60057. 문자열 압축/문자열 압축.java
import java.util.*;

class Solution {
    public int solution(String s) {
        int slength = s.length();
        
        for(int i=1; i<=s.length()/2; i++){
            int count =1;
            String result = ""; // 결과
            String pre = s.substring(0,i); // 자르기
            
            for(int j =i; j<s.length(); j+=i){
                String sub;
                
                if(j+i>s.length()){
                    sub = s.substring(j);
                }else {
                    sub = s.substring(j,j+i);
                }
                
                if(pre.equals(sub)){
                    count++;
                }else{
                    result += (count>1 ? count : "") + pre;
                    pre=sub;
                    count =1; // 초기화
                }
                
            }
            result += (count>1 ? count : "") + pre;
            
            slength = Math.min(slength,result.length()); // 최소값
        }
        
        return slength;
    }
}

File: 프로그래머스/2/60057. 문자열 압축/문자열 압축.py
def solution(s):
    # 반복되는 문자열 압축시키기
    if len(s) == 1 : return 1
    
    min_lenght = len(s)
    
    for step in range(1, len(s)//2 +1):
        slic = ""
        prev = s[:step]
        count =1
        
        for i in range(step, len(s), step):
            if prev == s[i:i+step]:
                count+=1
            else :
                if count > 1 :
                    slic += str(count)+prev
                else :
                    slic += prev
                count = 1
                prev = s[i:i+step]
                    
        if count > 1:
            slic += str(count)+prev
        else :
            slic += prev    
            
        min_lenght  = min(min_lenght , len(slic))
    
    return min_lenght 